pkg/cloudproviders/awsprovider/templates/controlPlane.yaml

```diff
@@ -0,0 +1,470 @@
+      GroupId: !Ref SecurityGroup
+      SourceSecurityGroupId: !Ref SecurityGroup
+      IpProtocol: -1
+
+  EC2Role:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service: ec2.amazonaws.com
+            Action: sts:AssumeRole
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/AmazonS3ReadOnlyAccess
+
+  InstanceProfile:
+    Type: "AWS::IAM::InstanceProfile"
+    Properties:
+      Roles:
+        - !Ref EC2Role
+
+  ControlPlaneLaunchTemplate:
+    Type: AWS::EC2::LaunchTemplate
+    Metadata:
+      AWS::CloudFormation::Authentication:
+        S3AccessCreds:
+          type: "S3"
+          roleName: !Ref EC2Role
+      AWS::CloudFormation::Init:
+        configSets:
+          init:
+            - repoKey
+        repoKey:
+          files:
+            "/root/attachHost.sh":
+              mode: '000100'
+              owner: root
+              group: root
+              source: !Ref IgnitionScriptS3URL
+              authentication: "S3AccessCreds"
+    Properties:
+      LaunchTemplateName: !Join ["-", [!Ref 'AWS::StackName', "ControlPlaneLaunchTemplate"]]
+      LaunchTemplateData:
+        BlockDeviceMappings:
+          - Ebs:
+              VolumeSize: 25
+              VolumeType: gp3
+              Iops: 3000
+              DeleteOnTermination: true
+              Encrypted: false
+            DeviceName: /dev/sda1
+          - Ebs:
+              VolumeSize: 100
+              VolumeType: gp3
+              Iops: 3000
+              DeleteOnTermination: true
+              Encrypted: false
+            DeviceName: /dev/sdb
+        ImageId: !Ref ImageID
+        InstanceType: !Ref InstanceType
+        IamInstanceProfile:
+          Name: !Ref InstanceProfile
+        SecurityGroupIds:
+          - !Ref SecurityGroup
+        UserData:
+          Fn::Base64:
+            Fn::Sub:
+              |
+              #!/bin/bash -xe
+              # install the packages required for cfn
+              sudo yum -y update
+              sudo yum -y install python3 python3-pip
+              pip3 install pystache
+              ln -s /usr/bin/python3 /usr/bin/python
+              # download the cfn-bootstrap script and install the package
+              sudo curl -O https://s3.amazonaws.com/cloudformation-examples/aws-cfn-bootstrap-py3-latest.tar.gz
+              sudo tar xvf aws-cfn-bootstrap-py3-latest.tar.gz
+              cd aws-cfn-bootstrap-*/
+              sudo python setup.py build
+              sudo python setup.py install
+              sudo chmod 775 /usr/local/init/redhat/cfn-hup
+              # download the ignition script to the instance
+              sudo /usr/local/bin/cfn-init -v --stack ${AWS::StackName} --region ${AWS::Region} --resource ControlPlaneLaunchTemplate --configsets init
+              # run the script
+              sudo sh /root/attachHost.sh
```

Comment:
- let's be explicit and use `sudo bash` unless `sh` is always `bash` in our machines